cmd/kubeadm/app/cmd/upgrade/plan.go

```diff
@@ -39,7 +41,11 @@ func NewCmdPlan(parentFlags *cmdUpgradeFlags) *cobra.Command {
 			err := runPreflightChecks(parentFlags.skipPreFlight)
 			kubeadmutil.CheckErr(err)
 
-			err = RunPlan(parentFlags)
+			if cfg.FeatureGates, err = features.NewFeatureGate(&features.InitFeatureGates, parentFlags.featureGatesString); err != nil {
```

Comment:
I would move this exact code from here and in apply to `upgrade.go` where `cfg` is first defined